components/slideshow.tsx

```diff
@@ -0,0 +1,91 @@
+'use client';
+
+import Image from 'next/image';
+import { useCallback, useEffect, useState } from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+export default function Slideshow({ images }: { images: string[] }) {
+  const [currentSlide, setCurrentSlide] = useState<number>(0);
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  }, [images.length]);
+
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === images.length - 1 ? 0 : prev + 1
+    );
+  }, [images.length]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      nextSlide();
+    }, 3000);
+
+    return () => clearInterval(intervalId);
+  }, [nextSlide]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (!hovered) return;
+      if (event.key === 'ArrowLeft') {
+        prevSlide();
+      } else if (event.key === 'ArrowRight') {
+        nextSlide();
+      }
+    };
+    window.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [hovered, prevSlide, nextSlide]);
+
+  return (
+    <div
+      className="group relative overflow-hidden"
```

Comment:
Don't panic, read my previous comment carefully,

> Make it `opacity-60` for not hovered and `opacity-100` for hovered. Leave the group thing.

Because with the group tag, the hover will apply when you hover on the div itself; which is wrong. You change the opacity only when hovering the arrow icon.

I originally suggested to use group because I didn't notice that you did it wrong initially.

Reply:
oh my bad